src/provider/handler/mappers/mapPostMediaRequest.js

```diff
@@ -0,0 +1,24 @@
+import axios from 'axios';
+
+export function mapPostMediaRequest(baseUrl) {
+  return function(post) {
+    return axios
+      .get(`${baseUrl}/wp-json/wp/v2/media/${post.featured_media}`)
+      .then(response => {
+        if (
+          response.status == 200 &&
+          response.data &&
+          response.data.media_details &&
+          response.data.media_details.sizes &&
+          response.data.media_details.sizes.thumbnail
+        ) {
+          return {
+            id: response.data.id,
+            image: response.data.media_details.sizes.thumbnail.source_url
+          };
+        } else {
+          return { id: -1 };
```

Comment:
why are you returning a negative id here ?
if you want to exclude this result without crashing the `Promise.all` I believe you can just return null

Reply:
I can't return null but I can return {id: null}

Reply:
If you return `{ id: null }` here, it means that where you run `Promise.all(mapPostMediaRequest)`, one of the mediaItems will be `{ id: null }`, which may cause an issue cause you're not ruling out this case in the response processing. If you `return null` then I think it would remove the faulty item out of the array that contains all the promises